share/smack/top.py

```diff
@@ -899,85 +979,133 @@ def verify_bpl(args):
-        command += ["/tryCTrace", "/noTraceOnDisk", "/printDataValues:1"]
-        command += ["/k:%d" % args.context_bound]
-        command += ["/useProverEvaluate"]
-        command += ["/timeLimit:%s" % args.time_limit]
-        command += ["/cex:%s" % args.max_violations]
-        command += ["/maxStaticLoopBound:%d" % args.loop_limit]
-        command += ["/recursionBound:%d" % args.unroll]
-        command += ["/bopt:proverOpt:O:smt.qi.eager_threshold=100"]
-        command += ["/bopt:proverOpt:O:smt.arith.solver=2"]
-        if args.solver == 'cvc4':
-            command += ["/bopt:proverOpt:SOLVER=cvc4"]
-        elif args.solver == 'yices2':
-            command += ["/bopt:proverOpt:SOLVER=Yices2"]
+        command = invoke_corral(args)
 
     elif args.verifier == 'symbooglix':
-        command = ['symbooglix']
-        command += [args.bpl_file]
-        command += ["--file-logging=0"]
-        command += ["--entry-points=%s" % ",".join(args.entry_points)]
-        command += ["--timeout=%d" % args.time_limit]
-        command += ["--max-loop-depth=%d" % args.unroll]
+        command = invoke_symbooglix(args)
 
     if args.verifier_options:
         command += args.verifier_options.split()
 
     verifier_output = try_command(command, timeout=args.time_limit)
-    verifier_output = transform_out(args, verifier_output)
-    result = verification_result(verifier_output, args.verifier)
+    return process_verifier_output(args, verifier_output)
 
-    if args.json_file:
-        with open(args.json_file, 'w') as f:
-            f.write(json_output_str(result, verifier_output, args.verifier))
 
-    if result in VResult.ERROR:
-        error = error_trace(verifier_output, args.verifier)
+def get_result(result):
+    return result
 
-        if args.error_file:
-            with open(args.error_file, 'w') as f:
-                f.write(error)
 
-        if not args.quiet:
-            print(error)
+def thread_verify_bpl(args, commands_to_add):
+    args.modular = False
 
-        if args.replay:
-            replay_error_trace(verifier_output, args)
-    print(result.message(args))
-    return result.return_code()
+    if 'verifier' in commands_to_add:
+        args.verifier = commands_to_add['verifier']
+    else:
+        args.verifier = 'corral'
+
+    if 'modular' in commands_to_add:
+        args.modular = commands_to_add['modular']
+
+    if 'verifier-options' in commands_to_add:
+        args.verifier_options = commands_to_add['verifier-options']
+
+    if 'solver' in commands_to_add:
+        args.solver = commands_to_add['solver']
+
+    if args.verifier == 'boogie' or args.modular:
+        command = invoke_boogie(args)
+
+    elif args.verifier == 'corral':
+        command = invoke_corral(args)
+
+    elif args.verifier == 'symbooglix':
+        command = invoke_symbooglix(args)
+
+    if args.verifier_options:
+        command += args.verifier_options.split()
+
+    return args, try_command(command, timeout=args.time_limit)
```

Comment:
This will work, but it is weird to be invoking `try command` from the return statement. So maybe just return `command` here and then run it on the other side.

Reply:
Sorry, I just noticed that you have to run try_command here. So then run it in a separate statement just before the return statement.